### ISSUE TITLE
Port kdl_parser_py to ROS 2

### ISSUE THREAD

**Issue by clalancette** [ros/kdl_parser#46]
There are two packages in this repository: `kdl_parser` and `kdl_parser_py`.  We've had a ROS 2 port for the first one for a long time, but we have never gotten around to porting `kdl_parser_py` to ROS 2.  This is a good first task for someone to pick up.  The rough steps as I see them:

1.  Make sure the code in the `kdl_parser_py` package is up-to-date with the fixes that have landed on the noetic-devel branch.
1.  Restructure the `kdl_parser_py` package to be a pure Python package and not use CMakeLists.txt at all (see https://github.com/ros2/demos/tree/2b94e70a06d9a0b37ba6b7d3405e5e2d1779e509/demo_nodes_py as one example of a pure Python package).
1.  Remove the AMENT_IGNORE.
1.  Add in the linters (again, see https://github.com/ros2/demos/tree/2b94e70a06d9a0b37ba6b7d3405e5e2d1779e509/demo_nodes_py/test for examples).
1.  Port the tests to ROS 2.
1.  Open the PR.

**Comment by clalancette** [comment]
> Can I work on this? I've already made a ROS 2 Humble compatible version.

There is already an open pull request for this in #55.  The thing that is holding that back is that `urdf_parser_py` isn't currently part of the ROS 2 core, but `kdl_parser` is.  In order to land this we need to do one of:

1.  Remove the dependency of `urdf_parser_py` from `kdl_parser_py`.  I'm not sure how feasible this is.
2.  Split this repository into two repositories, one containing just `kdl_parser`, and one containing `kdl_parser_py`.  The former would live in the ROS 2 core as it does now, and the latter would only contain `kdl_parser_py`, which could then comfortably depend on `urdf_parser_py`.
3.  Import `urdf_parser_py` into https://github.com/ros2/ros2/blob/rolling/ros2.repos

My vote would be for 1, if possible, but 2 if not.  @mjcarroll what are your thoughts?

**Comment by mjcarroll** [comment]
I don't think that (1) is very feasible, the `urdf_parser_py` looks to be non-trivial at a quick glance.  I would also be reluctant to pull another package into `ros2.repos`.

If someone thinks that 1 is possible, that would be an optimal path, but I would be fine with 2.

**Comment by clalancette** [comment]
All right.  See https://github.com/ros/kdl_parser/pull/89 and https://github.com/ros/kdl_parser_py/pull/1 for a start to splitting kdl_parser_py out of this repository.  Once those are approved and merged, I'll close this issue and move #55 over to the new repository, where we can continue iterating on it there.
